Source/JavaScriptCore/runtime/StringPrototypeInlines.h

```diff
@@ -1200,10 +1202,12 @@ ALWAYS_INLINE JSString* replace(VM& vm, JSGlobalObject* globalObject, JSValue th
     if (searchValue.inherits<RegExpObject>())
         RELEASE_AND_RETURN(scope, replaceUsingRegExpSearch(vm, globalObject, string, searchValue, replaceValue));
 
-    if (searchJSString && replaceJSString) {
-        if (JSString* result = tryReplaceOneCharUsingString<DollarCheck::Yes>(globalObject, string, searchJSString, replaceJSString))
-            return result;
-        RETURN_IF_EXCEPTION(scope, nullptr);
+    if constexpr (replaceMode == StringReplaceMode::Single) {
```

Comment:
I was not able to reproduce a case where tryReplaceOneCharUsingString would actually be executed for replaceAll, but I think it makes sense to put in this guard just in case.